### ISSUE TITLE
string cleanup may break code

### ISSUE THREAD

**Issue by bobvdk** [rubocop/rubocop#11601]
Rubocop (v1.45.1) will break an unusual string fragment

`foo = "xrun.cmd\\\""`

--------

## Expected behavior
`foo = "xrun.cmd\\\""`
should be cleaned to
`foo = 'xrun.cmd\"'`

## Actual behavior

`foo = "xrun.cmd\\\""`
gets cleaned to:
`foo ='xrun.cmd"'`

## Steps to reproduce the problem

Here is a short code fragment which can be used to demonstrate the issue
```
# before rubocop run foo and fum are the same
# after rubocop run foo and bar ar the same


foo = "xrun.cmd\\\""
bar = 'xrun.cmd"'
fum = 'xrun.cmd\"'

puts "foo is |#{foo}|"
puts "bar is |#{bar}|"
puts "fum is |#{fum}|"
```

## RuboCop version

```bobvdk_tnrlnx2> rubocop -V
1.45.1 (using Parser 3.2.1.0, rubocop-ast 1.26.0, running on ruby 2.7.5) [x86_64-linux]
```

**Comment by koic** [comment]
`"xrun.cmd\\\""` seems to be autocorrected to `'xrun.cmd\"'` as expected.

```ruby
foo = "xrun.cmd\\\""
bar = 'xrun.cmd"'
fum = 'xrun.cmd\"'

puts "foo is |#{foo}|"
puts "bar is |#{bar}|"
puts "fum is |#{fum}|"
```

```console
% bundle exec rubocop example.rb -A --enable-pending-cops
Inspecting 1 file
C

Offenses:

example.rb:1:1: C: [Corrected] Style/FrozenStringLiteralComment: Missing frozen string literal comment.
foo = "xrun.cmd\\\""
^
example.rb:1:7: C: [Corrected] Style/StringLiterals: Prefer single-quoted strings when you don't need string interpolation or special symbols.
foo = "xrun.cmd\\\""
      ^^^^^^^^^^^^^^
example.rb:2:1: C: [Corrected] Layout/EmptyLineAfterMagicComment: Add an empty line after magic comments.
foo = 'xrun.cmd\"'
^

1 file inspected, 3 offenses detected, 3 offenses corrected
```

```diff
-foo = "xrun.cmd\\\""
+# frozen_string_literal: true
+
+foo = 'xrun.cmd\"'
 bar = 'xrun.cmd"'
 fum = 'xrun.cmd\"'
```

**Comment by koic** [comment]
I will close it because it will not reproduce, but if you have steps to reproduce it with the latest RuboCop, please feel free to reopen it.
